database/sql/src/main/scala/cromwell/database/slick/JobStoreSlickDatabase.scala

```diff
@@ -1,30 +1,35 @@
 package cromwell.database.slick
 
-import cats.instances.future._
-import cats.syntax.functor._
 import cromwell.database.sql.JobStoreSqlDatabase
 import cromwell.database.sql.joins.JobStoreJoin
+import cromwell.database.sql.tables.JobStoreSimpletonEntry
 
 import scala.concurrent.{ExecutionContext, Future}
-import scala.language.postfixOps
 
 trait JobStoreSlickDatabase extends JobStoreSqlDatabase {
   this: SlickDatabase =>
 
   import dataAccess.driver.api._
 
-  private def addJobStore(jobStoreJoin: JobStoreJoin)(implicit ec: ExecutionContext): DBIO[Unit] = {
-    for {
-      jobStoreEntryId <- dataAccess.jobStoreEntryIdsAutoInc += jobStoreJoin.jobStoreEntry
-      _ <- dataAccess.jobStoreSimpletonEntryIdsAutoInc ++=
-        jobStoreJoin.jobStoreSimpletonEntries.map(_.copy(jobStoreEntryId = Option(jobStoreEntryId)))
+  override def addJobStores(jobStoreJoins: Seq[JobStoreJoin], batchSize: Int)
+                           (implicit ec: ExecutionContext): Future[Unit] = {
+
+    def assignJobStoreIdsToSimpletons(jobStoreIds: Seq[Int]): Seq[JobStoreSimpletonEntry] = {
+      val simpletonsByJobStoreEntry = jobStoreJoins map { _.jobStoreSimpletonEntries }
+      val jobStoreIdsAndSimpletons = jobStoreIds.toList zip simpletonsByJobStoreEntry
+      jobStoreIdsAndSimpletons flatMap { case (id, simpletons) => simpletons.map(_.copy(jobStoreEntryId = Option(id))) }
+    }
+
+    val jobStoreEntries = jobStoreJoins map { _.jobStoreEntry }
+
+    val action = for {
+      jobStoreIds <- dataAccess.jobStoreEntryIdsAutoInc ++= jobStoreEntries.toList
```

Comment:
As mentioned in the comments below, the bulk `++=` for job store entries works at the Slick API level, but inspecting the SQL shows that inserts aren't actually done in bulk.  This is presumably due to using the `jobStoreEntryIdsAutoInc` instead of the `TableQuery`, but it appears we require the former in order to get the job store IDs to assign to the job store simpletons.